### DISCUSSION TITLE
Run PBPopupControllerExample on mac catalyst

### DISCUSSION THREAD

**Discussion by iDevelopper** [LeoNatan/LNPopupController#489] [General] [answered]
**Describe the Bug**
Run on macOS crash.

**To Reproduce**
Run on macOS

**Expected Behavior**
No crash.

**Console**
dyld: Library not loaded: @rpath/LNPreviewToContextMenu.framework/Versions/A/LNPreviewToContextMenu
  Referenced from: /Users/patrickbodet/Library/Developer/Xcode/DerivedData/LNPopupControllerExample-avvnxmznqzdffbemvsiidzujeymg/Build/Products/Debug-maccatalyst/LNPopupControllerExample.app/Contents/MacOS/LNPopupControllerExample
  Reason: no suitable image found.  Did find:
	/Users/patrickbodet/Library/Developer/Xcode/DerivedData/LNPopupControllerExample-avvnxmznqzdffbemvsiidzujeymg/Build/Products/Debug-maccatalyst/PackageFrameworks/LNPreviewToContextMenu.framework/LNPreviewToContextMenu: code signature in (/Users/patrickbodet/Library/Developer/Xcode/DerivedData/LNPopupControllerExample-avvnxmznqzdffbemvsiidzujeymg/Build/Products/Debug-maccatalyst/PackageFrameworks/LNPreviewToContextMenu.framework/LNPreviewToContextMenu) not valid for use in process using Library Validation: mapped file has no cdhash, completely unsigned? Code has to be at least ad-hoc signed.
	/Users/patrickbodet/Library/Developer/Xcode/DerivedData/LNPopupControllerExample-avvnxmznqzdffbemvsiidzujeymg/Build/Products/Debug-maccatalyst/LNPopupControllerExample.app/Contents/MacOS/../Frameworks/LNPreviewToContextMenu.framework/LNPreviewToContextMenu: code signature in (/Users/patrickbodet/Library/Developer/Xcode/DerivedData/LNPopupControllerExample-avvnxmznqzdffbemvsiidzujeymg/Build/Products/Debug-maccatalyst/LNPopupControllerExample.app/Contents/MacOS/../Frameworks/LNPreviewToContextMenu.framework/LNPreviewToContextMenu) not valid for use in process using Library Validation: mapped file has no Team ID and is not a platform binary (signed with custom identity or adhoc?)
	/Users/patrickbodet/Library/Developer/Xcode/DerivedData/LNPopupControllerExample-avvnxmznqzdffbemvsiidzujeymg/Build/Products/Debug-maccatalyst/LNPopupControllerExample.app/Contents/MacOS/../Frameworks/LNPreviewToContextMenu.framework/LNPreviewToContextMenu: stat() failed with errno=1
	/Users/patrickbodet/Library/Developer/Xcode/DerivedData/LNPopupControllerExample-avvnxmznqzdffbemvsiidzujeymg/Build/Products/Debug-maccatalyst/PackageFrameworks/LNPreviewToContextMenu.framework/Versions/A/LNPreviewToContextMenu: code signature in (/Users/patrickbodet/Library/Developer/Xcode/DerivedData/LNPopupControllerExample-avvnxmznqzdffbemvsiidzujeymg/Build/Products/Debug-maccatalyst/PackageFrameworks/LNPreviewToContextMenu.framework/Versions/A/LNPreviewToContextMenu) not valid for use in process using Library Validation: mapped file has no cdhash, completely unsigned? Code has to be at least ad-hoc signed.
	/Users/patrickbodet/Library/Developer/Xcode/DerivedData/LNPopupControllerExample-avvnxmznqzdffbemvsiidzujeymg/Build/Products/Debug-maccatalyst/LNPopupControllerExample.app/Contents/MacOS/../Frameworks/LNPreviewToContextMenu.framework/Versions/A/LNPreviewToContextMenu: code signature in (/Users/patrickbodet/Library/Developer/Xcode/DerivedData/LNPopupControllerExample-avvnxmznqzdffbemvsiidzujeymg/Build/Products/Debug-maccatalyst/LNPopupControllerExample.app/Contents/MacOS/../Frameworks/LNPreviewToContextMenu.framework/Versions/A/LNPreviewToContextMenu) not valid for use in process using Library Validation: mapped file has no Team ID and is not a platform binary (signed with custom identity or adhoc?)
	/Users/patrickbodet/Library/Developer/Xcode/DerivedData/LNPopupControllerExample-avvnxmznqzdffbemvsiidzujeymg/Build/Products/Debug-maccatalyst/LNPopupControllerExample.app/Contents/MacOS/../Frameworks/LNPreviewToContextMenu.framework/Versions/A/LNPreviewToContextMenu: stat() failed with errno=1
(lldb) 

**Additional Context**
macOS Big Sur 11.6.1, Xcode 13.0

**Comment by LeoNatan** [comment]
Hello,
I don't know what "PBPopupControllerExample" is, but if you mean the example project in this repo, `LNPreviewToContextMenu` is an SPM dependency which Xcode needs to fetch and build. Usually, this is done when opening the project.

**Reply by iDevelopper** [reply]
Sorry for PB, I meant LN!

> LNPreviewToContextMenu is an SPM dependency which Xcode needs to fetch and build. Usually, this is done when opening the project.

This is the case. After cloning your project (git clone https://github.com/LeoNatan/LNPopupController.git --recurse-submodules) and opening it, Xcode fetch and build the three SPM dependencies but running on macOS crash as I said before.

**Reply by LeoNatan** [reply]
Sorry, not sure what is up in your case. For me it runs just fine.

**Reply by iDevelopper** [reply]
<img width="1418" alt="image" src="https://user-images.githubusercontent.com/1782450/139592712-3fecda21-f012-48c1-acdd-2291da8df238.png">

**Reply by LeoNatan** [reply]
Ahh I see, it’s codesigning issue in your case. Change the account to something different.

**Reply by LeoNatan** [reply]
It tells you in the error message.

**Reply by iDevelopper** [reply]
What do you mean by account? Team? It is set to mine.

**Reply by LeoNatan** [reply]
Something is broken with code signing on your end.

**Reply by LeoNatan** [reply]
To work around this, you can disable library validation.

**Reply by iDevelopper** [reply, marked as answer]
I changed that:
<img width="435" alt="image" src="https://user-images.githubusercontent.com/1782450/139593304-cc072138-bfc6-4a20-8107-d53262b8dbd0.png">

by that:
<img width="435" alt="image" src="https://user-images.githubusercontent.com/1782450/139593325-6eaabcb0-a6c8-425d-afea-dbfa5856eecc.png">

And it works now!

**Reply by LeoNatan** [reply]
Interesting.
